### ISSUE TITLE
[MAYA-124103] Memory consumption increase upon USD export but not reduced after it's done

### ISSUE THREAD

**Issue by msercheli** [Autodesk/maya-usd#2426]
**Describe the bug**
While exporting a heavy mesh (very high poly count) as USD using the Python command mayaUSDExport, the system memory consumption increase (which is expected) but after the USD stage is saved to disk, the memory consumption does not reduce. Repeating this process for many different objects will eventually cause the system to run out of memory. This doesn't happen with Alembic caching for instance.

**Steps to reproduce**
1. Load in Maya a heavy mesh (400,000 faces), or create one
2. From Script Editor (Python) run:
```
cmds.mayaUSDExport(
    selection=True,
    file=<path to a .usd file on disk>,
    frameRange=[1,100]
)
```
3. The number of frames to export and the number of vertices/faces is high enough to reveal the issue
4. Follow the progress using a System Monitor tool that shows memory usage

**Expected behavior**
After saving the USD stage, memory should be freed and go back to the amount before the process started.

**Attachments**
N/A

**Specs (if applicable):**
 - CentOS 7.9
 - Compiler gcc 6.3.1
 - Maya version 2022.3 and 2023
 - Maya USD commit: I observed this with tags 0.17, 0.18
 - Pixar USD commit: tag 21.08

**Additional context**
Currently, due to the nature of our projects, our characters are very high poly count (around 394,000 faces). Their rigs are also quite heavy (the rig Maya file is around 1Gb in file size) After referencing the rig into Maya, the memory consumption increases by 7 Gb. While exporting their meshes as USD data, we are observing memory increases of 10 Gb for 100 frames with 3 frame samples (300 USD time-samples). Without anything running (only Maya), repeating this process for some characters would eventually cause our workstations to run out of memory (we have in general a total of 64 Gb of RAM).
Although we need to optimize our rigs and potentially our meshes, the issue is still valid in my opinion.

**Comment by santosg87** [comment]
I am able to see the memory issue you mention. we are investigating a bit deeper. it seems that the memory is release on file new, Although i'm not sure this is a viable workaround. :)

**Comment by neilrubie** [comment]
I have seen the same memory-devouring issue in my testing and doing a file->new did nothing to release any memory. It only seems to get released on closing down Maya. (which is even less viable as a workaround...)

**Comment by santosg87** [comment]
this issue should be fixed now on MayaUSD 0.23.0 

marking this as closed for the time being. :)
